src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -0,0 +1,199 @@
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
+import java.util.function.BinaryOperator;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * A GATKCommandLinePluginDescriptor for the ReadFilter class
+ */
+public class GATKReadFilterPluginDescriptor extends GATKCommandLinePluginDescriptor<ReadFilter> {
+
+    @Argument(fullName = StandardArgumentDefinitions.READ_FILTER_LONG_NAME,
+            shortName = StandardArgumentDefinitions.READ_FILTER_SHORT_NAME,
+            doc="Read filters to be applied before analysis", optional=true)
+    public final List<String> readFilterNames = new ArrayList<>(); // preserve order
+
+    final String disableReadFilterArgName = "disableReadFilter";
+    @Argument(fullName = disableReadFilterArgName,
+            shortName = "df",
+            doc="Read filters to be disabled before analysis", optional=true)
+    public final Set<String> disableFilters = new HashSet<>();
+
+    // Map of read filter class (simple) names to the corresponding plugin instance
+    public Map<String, ReadFilter> readFilters = new HashMap<>();
+
+    /////////////////////////////////////////////////////////
+    // GATKCommandLinePluginDescriptor implementation methods
+
+    @Override
+    public Class<?> getPluginClass() {return org.broadinstitute.hellbender.engine.filters.ReadFilter.class;}
+
+    @Override
+    public String getPackageName() {return "org.broadinstitute.hellbender.engine.filters";};
+
+    @Override
+    public Predicate<Class<?>> getClassFilter() {
+        //TODO should we use an opt-in annotation instead of runtime filtering ?
+        return c -> {
+            // don't use the ReadFilter base class, or the CountingReadFilter, or the unit tests
+            return !c.getName().equals(this.getPluginClass().getName()) &&
+                    !c.getName().startsWith(this.getPluginClass().getName() + "$") &&
+                    !c.getName().startsWith(CountingReadFilter.class.getName()) &&
+                    !c.getName().contains("UnitTest$");
+        };
+    }
+
+    // Instantiate a new ReadFilter derived object and save it in the list
+    @Override
+    public Object addInstance(final Class<?> pluggableClass)
+            throws IllegalAccessException, InstantiationException {
+        final ReadFilter readFilter = (ReadFilter) pluggableClass.newInstance();
+        // use getSimpleName; the classes are all in the same package
+        readFilters.put(pluggableClass.getSimpleName(), readFilter);
+        return readFilter;
+    }
+
+    // Verify that this target plugin class's dependent value has been specified
+    @Override
+    public boolean isDependentArgumentAllowed(Class<?> targetPluginClass) {
+        return readFilterNames.contains(targetPluginClass.getSimpleName());
```

Comment:
Filters requested by the tool and filters excluded should be taken into account here (or validation has to be delayed somehow). May need to have the arg parsing system notify via callback whenever it encounters an argument for a particular plugin class, so that you can later determine whether the plugin args the user specified were appropriate given the final selection of plugins.